[Dominic Ng Jun Yao] iP

dom-buri opened the pull request:
### Dommi
> "Carry out a random act of kindness, with no expectation of reward, safe in the knowledge that one day someone might do the same for you.." – Princess Diana ([source](https://www.brainyquote.com/quotes/princess_diana_154326?src=t_random))

Dommi frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,
1. download it from [here](https://github.com/dom-buri/ip/releases/tag/A-Jar)
1. double-click it.
1. add your tasks.
1. let it manage your tasks for you :wink:

And it is <b>FREE!</b>

Features:

- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
##
If you Java programmer, you can use it to practice Java too. Here's the `main` method:

``` java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

dom-buri committed:
Implement initial skeletal version of chatbot

dom-buri committed:
Add echo command and exit command for chatbot

dom-buri committed:
Allow chatbot to store user input and display them

dom-buri committed:
Create task class to represent a task list

dom-buri committed:
Allow marking tasks as done and not done

dom-buri committed:
Create Todo Deadline and Event class to support multiple task types

dom-buri committed:
Allow tracking of tasks into different types

dom-buri committed:
Automate test of text UIs using I/O redirection technique

dom-buri committed:
Implement DukeException to handle chatbot errors

dom-buri committed:
Handle input errors entered by the user

dom-buri committed:
Change data structure for tasks list from array to arraylist for easier deletion of tasks

dom-buri committed:
Allow deletion of tasks from the list

dom-buri committed:
Use enums instead of string for user instructions

dom-buri committed:
Update test cases for new features

dom-buri committed:
Update .txt file when tasks are created or updated

dom-buri committed:
Update .txt file when tasks are deleted

dom-buri committed:
Read .txt file when chatbot starts up

dom-buri committed:
Merge branch 'branch-Level-7'

dom-buri committed:
Use Date class for storing of dates

dom-buri committed:
Merge branch 'branch-Level-8'

dom-buri committed:
Refactor using Ui, Parser and Storage classes

dom-buri committed:
Merge branch 'branch-MoreOOP'

dom-buri committed:
Organize classes into packages

dom-buri committed:
Merge branch 'add-gradle-support'

dom-buri committed:
Update main class name in Gradle

dom-buri committed:
Add JUnit tests for Parser and TaskList

dom-buri committed:
Create .jar file for distribution

dom-buri committed:
Implement searching of tasks

dom-buri committed:
Standardized code to given coding standard

dom-buri committed:
Add documentation to Duke

dom-buri committed:
Merge branch 'branch-A-JavaDoc'

dom-buri committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/data/task/Task.java

dom-buri committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/data/task/Task.java
#	src/main/java/duke/parser/Parser.java
#	src/main/java/duke/ui/Ui.java

bkjwjason commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,112 @@
+        case delete:
+            return new DeleteTaskCommand(Integer.parseInt(second_word));
+        case find:
+            return new FindCommand(second_word);
```
I like how you remembered the Java style for switch and case statements!

bkjwjason commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,85 @@
+
+public class Storage {
+    /** the path of the .txt file */
+    public static String DEFAULT_STORAGE_FILEPATH;
```
Great job adhering to naming conventions here for constants!

bkjwjason commented on src/test/java/duke/data/TaskListTest.java:
```diff
@@ -0,0 +1,71 @@
+public class TaskListTest {
+    /** Tests add feature of TaskList */
+    @Test
+    public void test_add() {
```
Should the test method names be in camelCase? Or are these an exception?

bkjwjason commented on src/test/java/duke/parser/ParserTest.java:
```diff
@@ -0,0 +1,74 @@
+
+    /** Tests list feature of Parser */
+    @Test
+    public void parse_list() throws DukeException, ParseException {
```
Same for these methods, wondering if test method names should be in camelCase as well?

bkjwjason commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,112 @@
+     * @throws IllegalArgumentException when the user enters an invalid command.
+     */
+    public static Command parseCommand(String command) throws ParseException, DukeException, IllegalArgumentException {
+        SimpleDateFormat inputDateFormatter = new SimpleDateFormat("yyyy-MM-dd");
```
Great job on using nouns to name your formatter!

bkjwjason reviewed (commented):
Overall, I found your code very easy to read as code was well structured and best practices were closely followed. I noted a minor naming deviation for your testing methods but I'm unsure if testing method names are an exception. Great job!

dom-buri committed:
Add checkStyle to detect coding style violations

dom-buri committed:
Fix coding style violations

dom-buri committed:
Merge branch 'branch-A-CheckStyle'

dom-buri committed:
Create GUI for chat bot

dom-buri committed:
Merge branch 'branch-Level-10'

erohsikivar commented on src/main/java/duke/data/task/Event.java:
```diff
@@ -0,0 +1,40 @@
+    }
+
+    @Override
+    public String toWrite() {
```
Great to see that you have used SimpleDateFormat class.

erohsikivar reviewed (commented):
Overall LGTM, maybe you consider creating a new package for your GUI.

dom-buri committed:
Generate new JAR file using Gradle

dom-buri committed:
Add assertions for Tasklist, Commands and Storage

dom-buri committed:
Test for assertions using Gradle

dom-buri committed:
Improve Code Quality

Parser: Abstract the splitting of user input into two strings
This makes the parseTask method reduce the number of level of
abstraction.

Duke: Remove the unused constructor
To clean up unused codes.

MainWindow: Remove unused variables

Commands: Remove unused method

dom-buri committed:
Merge pull request #1 from dom-buri/branch-A-CodeQuality

Improve Code Quality

dom-buri committed:
Merge branch 'master' into branch-A-Assertions

dom-buri committed:
Merge pull request #2 from dom-buri/branch-A-Assertions

Add assertions for Tasklist, Commands and Storage

dom-buri committed:
Set up CI

dom-buri committed:
Prevent newly created event to clash with existing events

dom-buri committed:
Merge branch 'branch-B-DetectAnomalies'

dom-buri committed:
Fix checkstyle errors

dom-buri committed:
Add OOP by creating gui package

dom-buri committed:
Add green bubble for text and adjust image size

dom-buri committed:
Update phrases used by Duke

dom-buri committed:
Handle empty description error description is empty for new todo

dom-buri committed:
Fix checkstyle errors

dom-buri committed:
Add test cases for Duke getResponse method

dom-buri committed:
Add User Guide

dom-buri committed:
Improve code quality of JUnit test methods

dom-buri committed:
Add screenshots to user guide

dom-buri committed:
Create new release v0.2

dom-buri committed:
Update user guide to include release v0.2 link

dom-buri committed:
Improve JavaDocs

dom-buri committed:
Add no tasks found for search

dom-buri committed:
Fix truncated text in GUI
